### PR TITLE
add createPerFrameHandlerObject

### PR DESCRIPTION
**Pull request by commy2**
**When merged this pull request will:**
- Reimplements `CBA_fnc_createPerFrameHandlerObject` (formerly known as `cba_common_fnc_addPerFrameHandlerLogic`)

Someone in the ACE public chat asked what happened to this. I guess this was meant to be API after all, so now I feel bad for nuking it. Should've been better documented though tbqh.

old one:
https://github.com/sickboy/cba/blob/542e28c71165864a32211a347a72b206d91ef882/addons/common/init_perFrameHandler.sqf#L58

Since this was removed before 2.4 I also took the opportunity to make some tweaks:
- Use a location instead of an object for minimal overhead.
- Switched around `_args` and `_delay` parameter to make it consistent with `addPerFrameHandler`
- The start code now also gets the logic as argument instead of the passed arguments directly. This makes it consistent with all the other called functions.
- And obviously removed `_common_` from the function name.
- `private keyword`, `pushBack` etc. etc.

### CODE REVIEW

**Comment by Glowbal** [comment]
I think the name `addPerFrameHandlerLogic` is strange and doesn't properly say what the function does. I'm not sure exactly what a better name should be yet.

**Comment by commy2** [comment]
`createPerFrameHandlerLogic` ?

**Comment by Glowbal** [comment]
It's the logic bit that makes it not fit, I think. Though create is indeed better.

**Comment by commy2** [comment]
Hmm. Other suggestion: `createPerFrameHandlerObject`. Although `Logic` is the term for dummy objects that do magical script things in Arma. (e.g. `class Logic;`)

**Comment by commy2** [comment]
I'm thinking about using these for my port of ACE2 fire propagation.

**Comment by Glowbal** [comment]
It is, but I see no reason to indicate we use logic objects in the function name. I will think of some suggestions for a better name.

**Comment by commy2** [comment]
The logic is a fundamental part of this though. It is the return value and the only passed argument in the called functions (aside from the `_private` functions).

**Comment by Killswitch00** [comment]
Object? Entity? Thing? Manager? Agent? (perhaps not, since the concept of "agents" exist in the game).

(Logic seems fine though)

**Comment by commy2** [comment]
`Object` would perfectly describe the character of this being. But the problem is that the game already has \<OBJECT\> as data type and this is actually a \<LOCATION\>...

**Comment by Killswitch00** [comment]
Runner? `CBA_fnc_createPerFrameHandlerRunner`

**Comment by Glowbal** [comment]
Object is the best of those so far. But you could also just leave it out and use `createPerFrameHandler`. 

Logic might be confused with some kind of internal function that handles the per frame handler logic.

**Comment by Killswitch00** [comment]
How about a remote/destroy/delete function as well?

File: addons/common/fnc_createPerFrameHandlerObject.sqf [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    if (isNil "_logic" || {isNull _logic}) exitWith {
+        (_logic getVariable "handle") call CBA_fnc_removePerFrameHandler;
+    };
+
+    // deserialize
+    private (_logic getVariable "private");
+    { call _x } forEach (_logic getVariable "deserialize");
+
+    // check exit condition - exit if false
```
Comment (Killswitch00) [inline addons/common/fnc_createPerFrameHandlerObject.sqf]: exit if true

File: addons/common/fnc_deletePerFrameHandlerObject.sqf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+
+params [["_logic", locationNull, [locationNull]]];
+
+if (!isNull _logic) then {
+    _logic call (_logic getVariable "end");
```
Comment (Killswitch00) [inline addons/common/fnc_deletePerFrameHandlerObject.sqf]: How about changing the handler's run and exit conditions to make it "end itself"?

**Comment by Killswitch00** [comment]
Looks about done now.

**Pull request merged**